### PR TITLE
Card component: remove unnecessary methods

### PR DESCRIPTION
**Pull request by jreij**

### CODE REVIEW

File: card/src/test/java/com/adyen/checkout/card/DefaultCardDelegateTest.kt [hunk 1]
```diff
@@ -331,9 +336,7 @@ internal class DefaultCardDelegateTest {
 
                 with(requireNotNull(expectMostRecentItem())) {
                     assertTrue(isValid)
-                    assertEquals(FieldState(TEST_CARD_NUMBER, Validation.Valid), cardNumberState)
-                    assertEquals(FieldState(TEST_SECURITY_CODE, Validation.Valid), securityCodeState)
-                    assertEquals(FieldState(TEST_EXPIRY_DATE, Validation.Valid), expiryDateState)
+                    assertEquals(createOutputData(), this)
```
Comment (jreij) [inline card/src/test/java/com/adyen/checkout/card/DefaultCardDelegateTest.kt:339]: @OscarSpruit ended up asserting the whole object when applicable (mostly in the valid scenarios)

Comment (OscarSpruit) [reply card/src/test/java/com/adyen/checkout/card/DefaultCardDelegateTest.kt:339]: nice 🙌

Comment (OscarSpruit) [reply card/src/test/java/com/adyen/checkout/card/DefaultCardDelegateTest.kt:339]: to make it even shorter 👀 
```Kotlin
assertEquals(createOutputData(), requireNotNull(expectMostRecentItem()))
```

Comment (jreij) [reply card/src/test/java/com/adyen/checkout/card/DefaultCardDelegateTest.kt:339]: There's an `assertTrue(isValid)` above it though

Comment (OscarSpruit) [reply card/src/test/java/com/adyen/checkout/card/DefaultCardDelegateTest.kt:339]: Ohhh sorry missed that one

**Review by OscarSpruit — APPROVED** [review]

**Pull request merged**